packages/number-input/src/Input/Input.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { ChangeEvent, useRef } from 'react';
+
+import { cx } from '@leafygreen-ui/emotion';
+import { useControlledValue } from '@leafygreen-ui/hooks';
+import Icon from '@leafygreen-ui/icon';
+import WarningIcon from '@leafygreen-ui/icon/dist/Warning';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { createSyntheticEvent } from '@leafygreen-ui/lib';
+
+import { Size, State } from '../NumberInput/NumberInput.types';
+
+import {
+  arrowBaseStyles,
+  arrowsAnimateStyles,
+  arrowsBaseStyles,
+  arrowThemeStyles,
+  errorInputStyles,
+  iconBaseStyles,
+  iconErrorStyles,
+  iconSizeStyles,
+  iconThemeStyles,
+  inputAnimateStyles,
+  inputBaseStyles,
+  inputThemeStyles,
+  selectBaseStyles,
+  sizeInputStyles,
+  warningIconClassName,
+  wrapperBaseStyles,
+  wrapperClassName,
+  wrapperDisabledStyles,
+  wrapperErrorStyles,
+  wrapperHoverStyles,
+  wrapperStateStyles,
+  wrapperThemeStyles,
+} from './Input.styles';
+import { InputProps } from './Input.types';
+
+/**
+ * @internal
+ */
+
+export function Input({
+  value: valueProp,
+  onChange: onChangeProp,
+  disabled,
+  size = Size.Default,
+  hasSelectOptions,
+  state = State.None,
+  errorMessage,
+  ...rest
+}: InputProps) {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const { theme } = useDarkMode();
+
+  const renderErrorIcon = state === State.Error;
+
+  const { value, handleChange: handleChangeProp } = useControlledValue(
```

Comment:
this naming is confusing... the value here is coming from a hook not necessarily from a prop, right? can we rename to controlledValue and controlledHandleChange?

Reply:
Alternatively, we can just not wrap the function in our own handleChange since it's not doing anything and we should just be able to call this `handleChange` directly

Reply:
yup!